frontend/packages/pipelines-plugin/src/components/pipelineruns/utils/tekton-results.ts

```diff
@@ -0,0 +1,335 @@
+              ? AND(
+                  ...expression.values.map((value) =>
+                    NEQ(`data.metadata.labels["${expression.key}"]`, value),
+                  ),
+                )
+              : '';
+          case 'In':
+            return expression.values?.length > 0
+              ? `data.metadata.labels["${expression.key}"] in [${expression.values.map(
+                  (value) => `"${value}"`,
+                )}]`
+              : '';
+          case 'Equals':
+            return expression.values?.[0]
+              ? EQ(`data.metadata.labels["${expression.key}"]`, expression.values?.[0])
+              : '';
+          case 'NotEquals':
+          case 'NotEqual':
+            return expression.values?.[0]
+              ? NEQ(`data.metadata.labels["${expression.key}"]`, expression.values?.[0])
+              : '';
+          case 'GreaterThan':
+            return expression.values?.[0]
+              ? EXP(`data.metadata.labels["${expression.key}"]`, expression.values?.[0], '>')
+              : '';
+          case 'LessThan':
+            return expression.values?.[0]
+              ? EXP(`data.metadata.labels["${expression.key}"]`, expression.values?.[0], '<')
+              : '';
+          default:
+            throw new Error(
+              `Tekton results operator '${expression.operator}' conversion not implemented.`,
+            );
+        }
+      })
+      .filter((x) => x),
+  );
+
+export const selectorToFilter = (selector) => {
+  let filter = '';
+  if (selector) {
+    const { matchLabels, matchExpressions, filterByName } = selector;
+
+    if (filterByName) {
+      filter = AND(filter, nameFilter(filterByName as string));
+    }
+
+    if (matchLabels || matchExpressions) {
+      if (matchLabels) {
+        filter = AND(filter, labelsToFilter(matchLabels));
+      }
+      if (matchExpressions) {
+        filter = AND(filter, expressionsToFilter(matchExpressions));
+      }
+    } else {
+      filter = labelsToFilter(selector as MatchLabels);
+    }
+  }
+  return filter;
+};
+
+// Devs should be careful to not cache a response that may not be complete.
+// In most situtations, caching is unnecessary.
+// Only cache a response that returns a single complete record as lists can change over time.
+let CACHE: { [key: string]: [any[], RecordsList] } = {};
+export const clearCache = () => {
+  CACHE = {};
+};
+const InFlightStore: { [key: string]: boolean } = {};
+
+// const getTRUrlPrefix = (): string => URL_PREFIX;
+
+export const createTektonResultsUrl = async (
+  namespace: string,
+  dataType: DataType,
+  filter?: string,
+  options?: TektonResultsOptions,
+  nextPageToken?: string,
+): Promise<string> => {
+  const tektonResult = await k8sGet(TektonResultModel, 'result');
+  const targetNamespace = tektonResult?.spec?.targetNamespace;
+  const route = await k8sGet(RouteModel, 'tekton-results-api-service', targetNamespace);
+  const tektonResultUrl = route?.spec.host;
```

Comment:
code means we make always two additional API calls when any TektonResults API call happen, right?

We could maybe have a local variable that contains the resultUrl and make the two API calls only when its not defined?

```ts
let tektonResultUrl: string;

export const createTektonResultsUrl () => {
  if (!tektonResultUrl) {
    const tektonResult = await k8sGet(TektonResultModel, 'result');
    const targetNamespace = tektonResult?.spec?.targetNamespace;
    const route = await k8sGet(RouteModel, 'tekton-results-api-service', targetNamespace);
    tektonResultUrl = route?.spec.host;
  }
}
```

wdyt?